packages/dialtone-vue2/recipes/item_layout/contact_info/contact_info.vue

```diff
@@ -26,23 +26,46 @@
             :full-name="avatar.fullName"
             :image-src="avatar.src"
             image-alt=""
-            :icon-name="avatarIcon"
-            :overlay-icon="avatar.icon"
             :overlay-text="avatar.text"
             :avatar-class="[{ 'd-mln24': index > 0, 'd-bc-brand': !!avatar.halo }]"
-          />
+          >
+            <template
+              v-if="avatarIcon"
+              #icon="{ iconSize }"
+            >
+              <component
+                :is="avatarIcon"
+                :size="iconSize"
+              />
```

Comment:
Shouldn't this just be a slot instead of expecting an icon component?

Reply:
It's because I cannot pass down a slot to the Avatar component. I didn't find any other way of doing this, please let me know if you have another solution.

Reply:
Yes I think dynamic rendering like this we could end up with similar tree shaking problems as we had with dt-icon .